cmd/boots/main.go

```diff
@@ -294,6 +298,34 @@ func getReporter(l log.Logger) (client.Reporter, error) {
 	}
 }
 
+// parseDynamicIPXEVars will parse any number of variable definitions from a
+// string, and return a an array of two-element arrays which are the key/value
+// string pairs of the variable's name and value. These will later be injected
+// as variable definitions in the iPXE script. Variable order is preserved.
+func parseDynamicIPXEVars(v string) ([][]string, error) {
+	if v == "" {
+		return nil, nil
+	}
+
+	// extract multiple variable definitions by splitting first on spaces
+	allDefs := strings.Fields(v)
```

Comment:
I'm wondering if we might run into problems with spaces within a field, where a quoted field value would be needed. 

https://ipxe.org/cfg -- scriptlet for example (https://forum.ipxe.org/printthread.php?tid=11058 has some examples)

https://stackoverflow.com/a/47489825/156674 has some suggestions over strings.Fields (not all of which apply in this situation, but offer some thought)

Reply:
Sounds like I could really use a unit test for this function and include some of these edge cases. I'll add that next.